core/jobs/io/gcs_io_test.py

```diff
@@ -0,0 +1,197 @@
+        content: bytes. The content to store.
+        generation: int. The file generation.
+
+    Returns:
+        file: FakeFile. The FakeFile that is stored to FakeGcs.
+    """
+    bucket, name = gcsio.parse_gcs_path(path)
+    file = gcsio_test.FakeFile(bucket, name, content, generation)
+    client.objects.add_file(file)
+    return file
+
+
+class ReadFileTest(job_test_utils.PipelinedTestBase):
+    """Tests to check gcs_io.ReadFile."""
+
+    def test_read_from_gcs(self) -> None:
+        client = gcsio_test.FakeGcsClient()
+        bucket = app_identity_services.get_gcs_resource_bucket_name()
+        gcs_url = f'gs://{bucket}/dummy_file'
+        string = b'testing'
+        insert_random_file(client, gcs_url, string)
+        filepaths = ['dummy_file']
+        filepath_p_collec = (
+            self.pipeline
+            | 'Create pcoll of filepaths' >> beam.Create(filepaths)
+            | 'Read file from GCS' >> gcs_io.ReadFile(client)
+        )
+        self.assert_pcoll_equal(filepath_p_collec, [string])
+
+
+class WriteFileTest(job_test_utils.PipelinedTestBase):
+    """Tests to check gcs_io.WriteFile."""
+
+    def test_write_to_gcs(self) -> None:
+        client = gcsio_test.FakeGcsClient()
+        string = b'testing'
+        filepaths = [
+            {
+                'filepath': 'dummy_folder/dummy_subfolder/dummy_file_1',
+                'data': string
+            },
+            {
+                'filepath': 'dummy_folder/dummy_subfolder/dummy_file_2',
+                'data': string
+            }
+        ]
+        filepath_p_collec = (
+            self.pipeline
+            | 'Create pcoll of filepaths' >> beam.Create(filepaths)
+            | 'Write to GCS' >> gcs_io.WriteFile(client)
+        )
+        self.assert_pcoll_equal(filepath_p_collec, [7, 7])
+
+
+class DeleteFileTest(job_test_utils.PipelinedTestBase):
+    """Tests to check gcs_io.DeleteFile."""
+
+    def test_delete_files_in_gcs(self) -> None:
+        client = gcsio_test.FakeGcsClient()
+        bucket = app_identity_services.get_gcs_resource_bucket_name()
+        gcs_url = f'gs://{bucket}/dummy_folder/dummy_subfolder/dummy_file'
+        string = b'testing'
+        insert_random_file(client, gcs_url, string)
+        file_paths = ['dummy_folder/dummy_subfolder/dummy_file']
+        filepath_p_collec = (
+            self.pipeline
+            | 'Create pcoll of filepaths' >> beam.Create(file_paths)
+            | 'Delete file from GCS' >> gcs_io.DeleteFile(client)
+        )
+        self.assert_pcoll_equal(filepath_p_collec, [None])
+
+    def _mock_delete(self, filepath: str) -> None:
+        """Mock function to check if files are passing correctly.
+
+        Args:
+            filepath: str. The filepath name.
+        """
+        bucket = app_identity_services.get_gcs_resource_bucket_name()
+        gcs_url = f'gs://{bucket}/dummy_folder/dummy_subfolder/dummy_file'
+        self.assertEqual(filepath, gcs_url)
+        return None
```

Comment:
You can move this into the `test_check_correct_files_are_passing` and you don't need to add a docstring.

Reply:
ditto elsewhere

Reply:
Done